### PR TITLE
BABEL-3802 Bigint/Int/Smallint/Tinyint Power Function

### PR DESCRIPTION
**Pull request by pratikzode**
At first in Babel, the Transact-SQL functions power always return float instead of 
returning the same type in SQL server.

In this commit we have added Bigint, Int, Smallint and Tinyint to power function.
Have added the test case to show Bigint, Int, Smallint and Tinyint to power.

Task: BABEL-3802
Signed-off-by: pratikzode <pratik.zode@gmail.com>

### CODE REVIEW

File: contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--2.3.0--2.4.0.sql [hunk 1]
```diff
@@ -49,6 +49,21 @@ CREATE OR REPLACE FUNCTION sys.degrees(IN arg1 TINYINT)
 RETURNS int AS 'babelfishpg_tsql','smallint_degrees' LANGUAGE C STRICT IMMUTABLE PARALLEL SAFE;
 GRANT EXECUTE ON FUNCTION sys.degrees(TINYINT) TO PUBLIC;
 
+CREATE OR REPLACE FUNCTION sys.power(IN arg1 BIGINT, IN arg2 FLOAT)
```
Comment (macwoody) [inline contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--2.3.0--2.4.0.sql]: I think you may need to modify a different upgrade file in 3x.dev , please double check

**Review by shalinilohia50 — DISMISSED** [review]

File: contrib/babelfishpg_tsql/runtime/functions.c [hunk 2]
```diff
@@ -1570,3 +1574,51 @@ smallint_radians(PG_FUNCTION_ARGS)
...
+Datum
+bigint_power(PG_FUNCTION_ARGS)
+{
+	int64	arg1 = PG_GETARG_INT64(0);
+	Numeric	arg2 = PG_GETARG_NUMERIC(1);
+	int64	result;
+	Numeric	arg1_numeric, result_numeric;
+
+	arg1_numeric = DatumGetNumeric(DirectFunctionCall1(int8_numeric,arg1));
+	result_numeric = DatumGetNumeric(DirectFunctionCall2(numeric_power, NumericGetDatum(arg1_numeric),Int64GetDatum (arg2)));
```
Comment (macwoody) [inline contrib/babelfishpg_tsql/runtime/functions.c]: for arg2, we still need to use NumericGetDatum

File: contrib/babelfishpg_tsql/runtime/functions.c [hunk 3]
```diff
@@ -1570,3 +1574,51 @@ smallint_radians(PG_FUNCTION_ARGS)
...
+Datum
+int_power(PG_FUNCTION_ARGS)
+{
+	int32	arg1 = PG_GETARG_INT32(0);
+	Numeric	arg2 = PG_GETARG_NUMERIC(1);
+	int32	result;
+	Numeric	arg1_numeric, result_numeric;
+
+	arg1_numeric = DatumGetNumeric(DirectFunctionCall1(int4_numeric,arg1));
+	result_numeric = DatumGetNumeric(DirectFunctionCall2(numeric_power, NumericGetDatum(arg1_numeric), Int32GetDatum (arg2)));
```
Comment (macwoody) [inline contrib/babelfishpg_tsql/runtime/functions.c]: same here

File: contrib/babelfishpg_tsql/runtime/functions.c [hunk 4]
```diff
@@ -1570,3 +1574,51 @@ smallint_radians(PG_FUNCTION_ARGS)
...
+
+	PG_RETURN_INT32(result); 
+}
+
+Datum
+smallint_power(PG_FUNCTION_ARGS)
+{
+	int16	arg1 = PG_GETARG_INT16(0);
+	Numeric	arg2 = PG_GETARG_NUMERIC(1);
+	int16	result;
```
Comment (macwoody) [inline contrib/babelfishpg_tsql/runtime/functions.c]: I remember we return int type, isn't it?

**Review by AvantikaDG — APPROVED** [review]

**Pull request merged**